core/src/main/resources/lib/layout/header/actions.jelly

```diff
@@ -20,17 +20,47 @@
 
     <!--  we only want the hamburger menu if there is something to display -->
     <!--  render all non primary non current actions -->
+    <j:set var="hamburgerHasBadgeWithSeverityError" value="false" />
+    <j:set var="hamburgerHasBadgeWithSeverityWarning" value="false" />
+    <j:set var="hamburgerHasBadgeWithSeverityInfo" value="false" />
     <j:set var="hamburgerEntries">
       <j:forEach var="action" items="${allActions}">
         <j:set var="isCurrent" value="${h.hyperlinkMatchesCurrentPage(action.urlName)}" />
         <j:set var="isPrimary" value="${h.showInPrimaryHeader(action)}"/>
         <j:if test="${!isCurrent and !isPrimary}">
           <h:secondaryAction action="${action}"/>
+          <j:set var="currentBadge" value="${action.badge}"/>
+          <j:if test="${currentBadge != null}">
+            <j:choose>
+              <j:when test="${currentBadge.severity == 'danger'}">
```

Comment:
i wonder whether the following may be shorter and easier to follow

```
<j:set var="hamburgerBadgeType" value="${currentBadge.severity}" />
...
<span class="jenkins-badge jenkins-!-${hamburgerBadgeType}-color" />
```
that assumes that there is always a severity, if not you would need to adopt to use a fallback value. the above in theory should get rid of the choose/when blocks

Reply:
I had a sync with James and my comment is not correct